### PR TITLE
Create minigame

### PR DESCRIPTION
**Pull request by Weldhappy**
a starter template...

### CODE REVIEW

**Comment by FelixFromDiscord** [comment]
It's not done, is it?

**Comment by Weldhappy** [comment]
idk

**Comment by Weldhappy** [comment]
no idea, but heres more..

-- MINIGAME: "Jewel Maze"

-- This minigame is inspired by the puzzle game Jewel Maze and takes place
on a grid of tiles. The player controls
-- a character and must navigate through the maze, collecting jewels and
reaching the exit.

-- SETUP

-- Create the player character
local player = {
  x = 8,
  y = 8,
  jewels = 0,
  sprite = "character"
}

-- Create a table to store the jewel objects
local jewels = {}

-- Create a 2D array to store the maze tiles
local maze = {}

-- Define the layout of the maze
local layout = {
  { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 },
  { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 }
}

-- Populate the maze with tiles and jewels
for y = 1, #layout do
  maze[y] = {}
  for x = 1, #layout[y] do
    if layout[y][x] == 1 then
      maze[y][x] = "wall"

-Scott

File: minigame [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  -- Update the player's position based on collisions with the ladder
+  if player.collidesWith(ladder1) then
+    if Keyboard.isDown("up") then
+      player.setY(player.getY() - 5)
+    elseif Keyboard.isDown("down") then
+      player.setY(player.getY() + 5)
+    end
+  end
+  
+  -- Update the player's position based on collisions with the platform
```
Comment (notiriscent) [inline minigame:80]: the code just cuts out here @Weldhappy

**Comment by Weldhappy** [comment]
crap, needs amendments and chatty g pitty

-Scott

**Pull request closed without merging**